core/src/main/scala/scalaz/FreeT.scala

```diff
@@ -107,6 +107,29 @@ sealed abstract class FreeT[S[_], M[_], A] {
   @deprecated("Alias for `interpret`", "7.3")
   def interpretT[T[_]](st: S ~> T)(implicit M: Functor[M]): FreeT[T, M, A] = interpret(st)
 
+  /**
+    * Runs to completion, mapping the suspension with the given transformation
+    * at each step and accumulating into the monad `M`.
```

Comment:
@TomasMikula a MonadRec could be useful but I think the effort might not be worth it.